Source/utils/string_or_view.hpp

```diff
@@ -39,9 +40,23 @@ class StringOrView {
 
 	StringOrView &operator=(StringOrView &&other) noexcept
 	{
+		string_view::size_type end = 0;

```

Comment:
`StringOrView` is a general class that is simply a union type between a `string` and a `string_view`. This behaviour is specific to `InfoString` and certainly doesn't belong here.

You can add a `SetInfoString(StringOrView&&)` (probably the easiest) function or add a separate class for `InfoString`.

Reply:
Do we need the view portion of StringOrView? I’m trying to make a new class that extends that one, but having some difficulty. Would it be okay if I made a new class based on std::string, or would that break something?

Reply:
Never mind. It would make several things fail to compile.